product_pricelist_supplierinfo/models/product_template.py

```diff
@@ -18,17 +18,23 @@ def _get_supplierinfo_pricelist_price(
         product = self.product_variant_id
         if product_id:
             product = product.browse(product_id)
-        if rule.no_supplierinfo_min_quantity:
-            quantity = 1.0
         # The product_variant_id returns empty recordset if template is not
         # active, so we must ensure variant exists or _select_seller fails.
         if product:
             if type(date) == datetime:
                 date = date.date()
-            seller = product._select_seller(
-                partner_id=rule.filter_supplier_id,
-                quantity=quantity,
-                date=date)
+            # If the field `no_supplierinfo_min_quantity` is set as True, we need
+            # to use the select seller that doesn't take into account the `min_qty`
+            # for the validation to know if the seller can be used.
+            if rule.no_supplierinfo_min_quantity:
+                seller = product._select_seller_without_min_qty(
```

Comment:
Why don't you inherit the original method and add that extra behavior?

Reply:
Because the original has the for cycle inside and it will return me the supplier info that is available, I can't add an extra validation, but I'm open to suggestions

Reply:
I mean, you may pass an extra parameter. If it's not set, fallback to the default behavior. If set, perform your custom stuf

Reply:
Ohh, oki, now is done